### PR TITLE
System settings and optional use of default locale

### PR DESCRIPTION
**Pull request by modsaid**
@frenkel This is one proposed solution for #179 to add the ability for more easy to manage configurations using settings.yml and a wrapper initializers/app_settings.rb

It is used to let the developers easily choose between the user-agent and the default locale for new non-logged-in requests

### CODE REVIEW

File: config/initializers/app_settings.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class AppSettings
+  def self.load
+    config_file = File.join(Rails.root, "config", "settings.yml")
+
```
Comment (modsaid) [inline config/initializers/app_settings.rb]: Trailing whitespace has been removed

File: config/initializers/app_settings.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      end
+    end
+  end
+
+  def self.method_missing(*args)
+    nil
+  end
+end
+AppSettings.load
+
```
Comment (modsaid) [inline config/initializers/app_settings.rb]: Trailing empty line has been removed

File: config/initializers/app_settings.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        send("#{key}=", config[key])
+      end
+    end
+  end
+
+  def self.method_missing(*)
+    nil
+  end
+end
+AppSettings.load
```
Comment (modsaid) [inline config/initializers/app_settings.rb]: has been added

**Comment by modsaid** [comment]
@frenkel  what do you think?

**Comment by frenkel** [comment]
Hi, sorry, I'm pretty busy these days, but it looks fine at first sight.
I'll have a better look near the end of the week.

One thing we ourselves are considering is adding an interface for configuration options like this, which would require storing them in the database. What are your thoughts about that?

**Comment by modsaid** [comment]
App settings that are stored in the database and managed through the admin users is also a  valid option for app  configuration.  I usually use both (file based and DB based)

The question is, which configuration to place on files, and which in the database?

I believe that configurations that should be stored in the database are the ones that are expected to changed by the agent (or application admin) upon desire and without involvement of developers or deployments.  (Activating Registration or not  is a good example)

File based configurations should be the ones that are tied to development and business logic.  (example: 3rd party API url)...  

In my opinion, the proposed switch here (to force a default language, regardless of the user browser language) falls in the latter file based category

**Comment by frenkel** [comment]
You're right, but for our hosted Brimir instances at https://getbrimir.com/ we give the user no access to files, so for them it would be better to have this as database configuration.
I want to merge your change, because it is better than having no configuration option for this at all. Could you add instructions to the README.md file how people can use this and what it does?

**Comment by modsaid** [comment]
For a hosted service it is definitely needed in the database.

Perhaps we can add it to the database configuration group when we introduce database based configuration.  and make it override the file based one.

For now, i'll add notes to the README

File: README.md [hunk 4]
```diff
@@ -43,6 +43,12 @@ If you want to use LDAP, configure config/ldap.yml accordingly, then change the
     bin/rails console production
     u = User.new({ email: 'your@email.address', password: 'somepassword', password_confirmation: 'somepassword' }); u.agent = true; u.save!
 
+Customization
+-------------
+Some applicant level configuration can be set through `config/settings.yml`
+
+Brimir is available in several languages. By default, it will use the locale corresponding to the user browser agent, if it was among the supported locales,. If you want to change this and force certain locale, you can do that by setting:   `ignore_user_agent_locale: true`  in  `config/settings.yml`
```
Comment (frenkel) [inline README.md]: `,.` should be `.` in here

**Comment by frenkel** [comment]
Thanks for you contribution. I've merged it all!

**Comment by modsaid** [comment]
Thank you @frenkel

**Pull request closed without merging**